### PR TITLE
feat(`crosschain`): add gas price increase limit 

### PR DESCRIPTION
**Pull request by lumtis**
# Description

Add a value `gasPriceIncreaseMax` in `GasPriceIncreaseFlags` to control the max gas price. The value is represented in percent of the median gas price for the chain

Set the default gas price max at 500% of the median gas price --> value to agree on

Closes: https://github.com/zeta-chain/node/issues/1313

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: proto/observer/crosschain_flags.proto [hunk 1]
```diff
@@ -13,6 +13,10 @@ message GasPriceIncreaseFlags {
     (gogoproto.stdduration) = true
   ];
   uint32 gasPriceIncreasePercent = 3;
+
+  // Maximum gas price increase in percent of the median gas price
+  // Default is used if 0
+  uint32 gasPriceIncreaseMax = 4;
```
Comment (ws4charlie) [inline proto/observer/crosschain_flags.proto:19]: Do we differentiate EVM-chain from BTC-chain on default `gasPriceIncreaseMax`

Comment (lumtis) [reply proto/observer/crosschain_flags.proto:19]: Right now `GasPriceIncreaseFlags` is general for all chains. This is why we use percentage, the `medianGasLimit` will be specific for each chains so the final max number will be different for each chains.

For Bitcoin, gas price increase is disabled.

Comment (ws4charlie) [reply proto/observer/crosschain_flags.proto:19]: Sounds cool. Thanks.

File: x/crosschain/keeper/abci.go [hunk 2]
```diff
@@ -85,6 +84,23 @@ func (k Keeper) CheckAndUpdateCctxGasPrice(
 	}
 	gasPriceIncrease := medianGasPrice.MulUint64(uint64(flags.GasPriceIncreasePercent)).QuoUint64(100)
 
+	// compute new gas price
+	currentGasPrice, err := cctx.GetCurrentOutTxParam().GetGasPrice()
+	if err != nil {
+		return math.ZeroUint(), math.ZeroUint(), err
+	}
+	newGasPrice := math.NewUint(currentGasPrice).Add(gasPriceIncrease)
```
Comment (ws4charlie) [inline x/crosschain/keeper/abci.go:92]: Is it possible that this `currentGasPrice` is outdated and significantly deviates (higher or lower) from the `medianGasPrice`? Are we `adjusting` the "gasPrice" based on (by percentage) the `currentGasPrice` or directly based on the dynamic `medianGasPrice`?

**Review by ws4charlie — APPROVED** [review]

**Review by brewmaster012 — APPROVED** [review]

File: x/crosschain/keeper/abci.go (hunk 2 above)
Comment (lumtis) [reply x/crosschain/keeper/abci.go:92]: `currentGasPrice` can be higher than `medianGasPrice` if it has already been increased by the same method previously. The gas price increase is calculated from the median gas price.

**Pull request merged**